### PR TITLE
range : fix and rework tooltips

### PR DESCRIPTION
**Pull request by AlicVB**
Following multiple issues (positionning and gtk_critical) see #12788 for ex.
Here is a rework that would simplify the code and remove the issues.

For instance, we have the following help "tooltips" for range widgets : 
- a "classic" tooltip under the widget, with "static" help infos + the current select, except for range rating
- a "popup" windows above the widget, which follow the mouse position and show the current value under the mouse + the current selection for range rating

But both have issues, inherent to gtk implementation (so we can't really change them) : 
- tooltips have an hardcoded delay before they are shown, and don't really follow the mouse pointer
- if we don't have enough space above the widget (topbar) we need to relocate the popup on the right, which cause flickering on start, because we can't know the popup height before it is shown
- another problem when relocating is that the resizing of the popup cause some GTK_CRITICAL (this looks like a gtk bug)

So what I propose here is to : 
- remove the "classic" tooltip
- move all information inside the "popup" that follow mouse position
- relocate the popup under the widget by default. And in the (rare) case where there's not enought space, it will automatically appears above, but that doesn't cause any GTK_CRITICAL, because the popup itself doesn't change it's size internally...
- One consequence is that we can easily have the same display (current value | current selection on top. which is good for consistency ;)

here's a screenshot : 
Capture d’écran_2022-11-13_15-26-53

Basically, it's in fact just a merge of both tooltips window

### CODE REVIEW

**Comment by TurboGit** [comment]
I don't think we need the filter label (**ISO** in your screenshot) inside the menu. We know that we are working on the ISO filter.

**Comment by AlicVB** [comment]
> I don't think we need the filter label (**ISO** in your screenshot) inside the menu. We know that we are working on the ISO filter.

hmm... actually not in the topbar (that's one of the issue with range in topbar)
and even in lateral module, it's not so obvious if the value in the first line doesn't speak by itself. For example it will be difficult to distinguish between different date filters.
So considering the height of this "tooltip" is not really big and won't grow further, I think that the redundancy here is better than the space gain ;)
... but as always it's just my opinion. I won't oppose if you are strongly in favour of that removing !

Note also that currently the tooltips show this information (in fact I haven't change the text)

**Comment by TurboGit** [comment]
> For example it will be difficult to distinguish between different date filters.

You're right of course! I though only about the collection filters place.

**Comment by dterrahe** [comment]
I'm fine with this approach. The only thing this "breaks", is that currently tooltips are automatically extended with shortcut information (if there is one or in mapping mode). But since at the moment shortcuts for the toolbar don't work correctly anyway (they only "work" in the lighttable, but really its the side panel shortcuts that work. In the darkroom the tooltip is showing any assigned shortcuts but they don't work because the side panel module isn't there).

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by dterrahe** [comment]
@AlicVB When you move over a date filter, the popover jitters as the dates continuously change, because the number characters are all different sizes. You'll want to add something like this (taken from gtk.c where this is applied to the toast to combat the same effect when moving sliders via shortcuts)
```
  PangoAttrList *attrlist = pango_attr_list_new();
  PangoAttribute *attr = pango_attr_font_features_new("tnum");
  pango_attr_list_insert(attrlist, attr);
  gtk_label_set_attributes(GTK_LABEL(darktable.gui->ui->toast_msg), attrlist);
  pango_attr_list_unref(attrlist);
```

**Comment by AlicVB** [comment]
@dterrahe : I suppose it depends of the font, because here the size doesn't change... I can see this for example on the lateral module, when hovering the widget on the left part of the range : when you move the mouse, the date change, but the "current selection" part doesn't move...
That said, your proposal doesn't seems bad... it's just that I can't test it ! have you try it on the range widgets ? if yes and if it works, then I' will do you the PR ;)
